src/routes/safe/components/Apps/components/ConfirmTxModal/ReviewConfirm.tsx

```diff
@@ -75,6 +79,8 @@ export const ReviewConfirm = ({
   const nativeCurrency = getNativeCurrency()
   const explorerUrl = getExplorerInfo(safeAddress)
   const isOwner = useSelector(grantedSelector)
+  const { remoteSafeApps } = useRemoteSafeApps()
```

Comment:
You don't need this. The modal takes an `app` prop that has an id.

Reply:
That id is not the id of type number that we are looking for. It is something of the type `{"url":"https://dapp.daohaus.club/","name":"DAOhaus"}`

as `SafeApp` type is https://github.com/gnosis/safe-react/blob/dda7f419599661d828165c111bbe58506c2ac5ca/src/routes/safe/components/Apps/types.ts#L4-L9

Reply:
The number id is overwritten

Reply:
What is the difference in the `id`s? Can we not just reference the 'new' `id` instead then?

Reply:
This is the overwritten id 
> That id is not the id of type number that we are looking for. It is something of the type {"url":"https://dapp.daohaus.club/","name":"DAOhaus"}

Reply:
Can we then pass the right id as a prop? I find it weird that the modal should fetch the list of apps again just to find an id.

Reply:
I have misunderstood your comment @katspaugh !! It makes all sense that we don't fetch the remote apps again so I prop drilled the appId from `<AppFrame />` . 👍